dive_core/perf_metrics_data.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ Copyright 2025 Google LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#include "dive_core/perf_metrics_data.h"
+
+#include <cctype>
+#include <cerrno>
+#include <cstdlib>
+#include <iostream>
+#include <filesystem>
+#include <string>
+#include <vector>
+#include <array>
+
+#include "dive_core/available_metrics.h"
+#include "dive_core/common/string_utils.h"
+
+namespace Dive
+{
+namespace
+{
+constexpr std::array kFixedHeaders = { "ContextID",   "ProcessID", "FrameID",
+                                       "CmdBufferID", "DrawID",    "DrawType",
+                                       "DrawLabel",   "ProgramID", "LRZState" };
+
+bool ParseHeaders(const std::string&              line,
+                  const AvailableMetrics&         available_metrics,
+                  std::vector<std::string>&       metric_names,
+                  std::vector<const MetricInfo*>& metric_infos)
```

Comment:
Optional: In the future, consider whether you can return the names and infos instead of using output parameters. This Abseil tip goes into more details: https://abseil.io/tips/176

 This can be accomplished using a struct wrapped in an optional:
```suggestion
struct ParseHeadersResult {
    std::vector<std::string> metric_names;
    std::vector<const MetricInfo*> metric_infos;
};
std::optional<ParseHeadersResult> ParseHeaders(const std::string& line, const AvailableMetrics& available_metrics)
```